### ISSUE TITLE
Test if C grid produces symmetric results

### ISSUE THREAD

**Issue by JFLemieux73** [CICE-Consortium/CICE#834]
./cice.setup --case Csym -g gbox80 -m ppp6 -e intel -p 1x1 -s boxforcee,boxclosed,buildincremental

I then changed things in the namelist:
npt=1 (days)
ndte = 2400
elasticDamp = 0.1d0
deltaminEVP = 2e-9
capping_method = 'sum'
kridge = -1
ktransport = -1
Pstar           = 1e4
ice_data_type = 'uniform'
ice_data_conc = 'p8'
ice_data_dist = 'uniform'
grid_ice     = 'C'

**Comment by JFLemieux73** [comment]
Finally I will use gbox12 instead...this is simpler. Let's start with atm_data_type   = 'uniform_east'. With gbox12, the first ocean T cell is 4,4 and the last one is 11,11. The four non-zero uvelE closest to the corners are 4,4 4,11 11,11 and 11,4. uvelE(4,11) and uvelE(4,4) should be equal (corolis=zero). Same idea for uvelE(10,11) and uvelE(10,4). I get the following values for the left and right sides after one solve (istep1=1) of the momentum eq:

velo istep1=           1
 velo left  7.017679076911179E-002  7.017679076911179E-002
 velo right  7.009409464899724E-002  7.009409464899724E-002

The values are symmetric and BFB. 

However after 24 time steps I get the following:

velo istep1=          24
 velo left  7.576958465637738E-002  7.576958465637798E-002
 velo right  7.572652350963513E-002  7.572652350963545E-002

The values are very close but not exactly equal.

**Comment by JFLemieux73** [comment]
I switch the wind direction: atm_data_type = 'uniform_west'. Now we get:

velo istep1=           1
 velo left -7.009409464899724E-002 -7.009409464899724E-002
 velo right -7.017679076911179E-002 -7.017679076911179E-002

They are exactly the same as above but with the minus sign. 

After 24 time steps:

 velo istep1=          24
 velo left -7.572652350963545E-002 -7.572652350963516E-002
 velo right -7.576958465637855E-002 -7.576958465637901E-002

**Comment by JFLemieux73** [comment]
Is it possible that the small differences  are due to the -O2 optimization? I tried by removing the flag or using instead -O0. The results are the same: the values are very close but not BFB.

**Comment by JFLemieux73** [comment]
Now setting  atm_data_type = 'uniform_north'.

and printing:

print *, 'velo bottom', vvelN(4,4,1), vvelN(11,4,1)
print *, 'velo top', vvelN(4,10,1), vvelN(11,10,1)

With istep1=1:

velo istep1=           1
 velo bottom  7.017679076911179E-002  7.017679076911179E-002
 velo top  7.009409464899724E-002  7.009409464899724E-002

These values are exactly the same as obtained for uvelE with atm_data_type = 'uniform_east'.

After 24 time steps:

velo istep1=          24
velo bottom  7.576958465637958E-002  7.576958465637741E-002
velo top  7.572652350963513E-002  7.572652350963513E-002

Interestingly this time the top values are BFB. The bottom values are very close but not BFB. 

I also tested atm_data_type = 'uniform_south' and the conclusions remain the same.

**Comment by JFLemieux73** [comment]
Now I set atm_data_type = 'uniform_northeast'.

In this case uvelE(10,11,1) should be equal to vvelN(11,10,1). Same idea for uvelE(4,4,1) and vvelN(4,4,1). 

print *, 'velo top   right', uvelE(10,11,1), vvelN(11,10,1)
print *, 'velo bottom left', uvelE(4,4,1), vvelN(4,4,1)

After one time step:
velo istep1=           1
velo top   right    8.812045377954515E-002  8.812045377954515E-002
velo bottom left  8.813531952509343E-002  8.813531952509343E-002

They are BFB. 

After 24 time steps:
Hum...in this case they are BFB:
velo istep1=          24
velo top   right  9.073835630042910E-002  9.073835630042910E-002
velo bottom left  9.073443752631669E-002  9.073443752631669E-002

The other cases like atm_data_type = 'uniform_southwest' do not exist. I modified the code to create these. 

With atm_data_type = 'uniform_southwest' :

velo istep1=          24
 velo top   right -9.073443752631669E-002 -9.073443752631669E-002
 velo bottom left -9.073835630042910E-002 -9.073835630042910E-002

BFB and same values (with minus sign) as with 'uniform_northeast'.

I also tested 'uniform_northwest' and 'uniform_southeast'. The results are the same (i.e. symmetric and BFB).

**Comment by JFLemieux73** [comment]
Conclusions from these simple box test cases:

-results are very close but not exactly BFB in order to get perfect symmetry when the wind blows in one direction.
-results are BFB symmetric when |uwind| = |vwind|

**Comment by eclare108213** [comment]
Interesting.  Thanks, @JFLemieux73, for such a thorough investigation.  These are close enough that I'm not too worried about them all being exactly BFB.  At some point, if/when one of us tries to make the B- and C-grid codes perfectly BFB, this will be very helpful for tracking down the differences.  When I did that for B-grid in the past, I had to make sure that every calculation in every direction was done in exactly the same order, i.e. as you rotate around the grid cell with the forcing, the order of terms appearing (left, right neighbors or whatever) have to be exactly the same under reflection.  It is interesting that the results are BFB symmetric when |uwind| = |vwind|.  That provides a lot of confidence that the implementation is correct, in addition to all the other tests you've done.  Looks good!
